src/module/entities/helper.js

```diff
@@ -369,72 +389,46 @@ module.exports = class UserProjectsHelper {
 					}
 				}
 
-				// Construct the filter to retrieve entity type IDs based on child hierarchy paths
-				const entityTypeFilter = {
-					name: {
-						$in: filteredHierarchyPaths,
-					},
+				// Construct the filter
+				const roleFilter = {
+					entityType: CONSTANTS.common.SUBROLE_ENTITY_TYPE,
```

Comment:
@priyanka-TL can we pass this entity type in request, if not explicitly passed in request  consider CONSTANTS.common.SUBROLE_ENTITY_TYPE,
because for this use case entityType of role is  "professional_subroles" if someone want to define another entity type and levels for role we can accommodate that as well